src/CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+# Check for necessary symbols and headers
+include(CheckSymbolExists)
+include(CheckIncludeFile)
+
+function(chkdef func inc)
+	string(TOUPPER ${func} FUNC)
+	set(HAVENAME HAVE_${FUNC})
+	check_symbol_exists(${func} "${inc}" ${HAVENAME})
+	if (${HAVENAME})
+		add_compile_definitions(${HAVENAME})
+	endif()
+endfunction()
+
+function(chkhdr inc req)
+	string(MAKE_C_IDENTIFIER ${inc} ID)
+	string(TOUPPER ${ID} INC)
+	set(HAVENAME HAVE_${INC})
+	check_include_file(${inc} ${HAVENAME})
+	if (${HAVENAME})
+		add_compile_definitions(${HAVENAME})
+	endif()
+	if (${req} AND NOT ${HAVENAME})
+		message(FATAL_ERROR "required include ${inc} not found")
+	endif()
+endfunction()
+
+chkdef(stpcpy string.h)
+chkdef(strerror string.h)
+chkdef(getuid unistd.h)
+chkdef(geteuid unistd.h)
+chkdef(mtrace mcheck.h)
+chkdef(secure_getenv stdlib.h)
+chkdef(__secure_getenv stdlib.h)
+chkdef(setreuid unistd.h)
+chkdef(setuid unistd.h)
+chkdef(stpcpy string.h)
+chkdef(strerror string)
+chkdef(vasprintf stdio)
+chkdef(srandom stdlib.h)
+chkdef(glob_pattern_p glob.h)
+chkdef(mbsrtowcs wchar.h)
+
+
+set(OPTINCS
+	fnmatch.h glob.h langinfo.h libintl.h mcheck.h stdalign.h
+)
+foreach(inchdr ${OPTINCS})
+	chkhdr(${inchdr} FALSE)
+endforeach()
+
+
+# Build the library
+set (popt_SOURCES
```

Comment:
I know this is how it was in my version and initially in the rpm cmake build as well. I don't find the style *offensive*, but AIUI these sources-as-variables kind of this isn't really in the cmake ethos, and I later changed rpm to just set these in the add_library() call - having them as variables doesn't buy anything at all.

Reply:
I thought I was going to use it in more places, but now that I don't, I'll merge them into the `add_library()` call.

Reply:
Yup, coming from automake background, I had similar feelings about it. Turns out it's unnecessary, you can add sources dynamically (not that it's needed for popt) natively in cmake and also get the sources from the target if necessary.

Reply:
This is fixed now.